### ISSUE TITLE
Sxiv-like shell script modularity

### ISSUE THREAD

**Issue by Q3CPMA** [eXeC64/imv#91]
Hello, and sorry if imv gets compared a lot to sxiv, but I wonder if it were feasible for imv to get something similar to sxiv's `.config/sxiv/exec/image-info` and `.config/sxiv/exec/key-handler`.
At least for me, it's the only thing preventing me from switching totally (especially since imv supports a lot more formats, and replaces hsetroot too).

**Comment by czarkoff** [comment]
What do you want it for?

**Comment by eXeC64** [comment]
What does `image-info` do? From the name I'm guessing `file(1)` could be a viable alternative.

As for `key-handler`, I'm assuming that means support for custom key bindings, and calling external scripts with optional variables such as the image's path, etc.

I've been meaning to implement support for config files (for things like default size, scaling mode, etc.), so it doesn't seem too far fetched to consider custom keybinding support.

Could you clarify what exactly your needs are?

**Comment by czarkoff** [comment]
The `key-handler` mechanism is redundant to `p` functionality, so I would argue that it really should not be implemented until there is a strong case for it.

@eXeC64: [sxiv's example `image-info` file](https://github.com/muennich/sxiv/blob/master/exec/image-info) adds file size to the description string. I'd rather implement it as a format string, to save some cpu cycles.

**Comment by eXeC64** [comment]
I agree, a format string should be perfectly adequate for the description, and the `p` bind is pretty flexible as it stands, so using that is definitely preferred to implementing custom key bindings.

I'd like to know what @Q3CPMA 's use case is still.

**Comment by Q3CPMA** [comment]
Yeah, image-info as a format string seems even better than sxiv's way, but I don't see how the `p` keybind can replace `key-handler`
I personally use it to delete files, rename them using dmenu or do some imagemagick thingies on the current one.

**Comment by eXeC64** [comment]
That does seem quite useful. This could be achieved by implementing #57 and including in the `[input]` section a `shell` command, e.g. `D = shell rm $path`

I've got a local branch with the start of an implementation that I ran out of time for. I've just finished my dissertation at university though, so I have more free time.

**Comment by Q3CPMA** [comment]
Would be even better than sxiv, since sxiv requires a prefix before the actual command.
Just hope this will permit full fledged shell commands (including pipes).
My key-handler actually has
```
case "$1" in
		"d")
			rm -- "$file" ;;
		"r")
			mv -n -- "$file" "$(dirname "$file")/$(ls "$(dirname "$file")" | dmenu -i -p "Rename $(basename "$file")")" ;;
	esac
```

**Comment by czarkoff** [comment]
Deleting files with `imv`:
```bash
czarkoff@e325:~/tmp/i $ ls
1.png 2.png 3.png 4.png
czarkoff@e325:~/tmp/i $ imv * | xargs rm
# p → p → → p q
czarkoff@e325:~/tmp/i $ ls
3.png
```
I'd say this way sucks much less then `exec()`ing scripts.

**Comment by Q3CPMA** [comment]
But I might be using imv from a file manager. And I might want to do more complicated things (how do I do the interactive rename, for example?).
I understand the desire to keep the tool pretty Unix-y and clean, but extensiveness isn't bloat.

**Comment by czarkoff** [comment]
@Q3CPMA: Anyway, this bug report covers 2 unrelated feature requests.  Could you please file them separately?

**Comment by eXeC64** [comment]
The problem with `p` is that it only allows one operation at a time when running imv. I like the flexibility the ability to configure a shell command binding would give. I don't want to call `exec()` on every keypress though. Internal configuration that only runs `exec()` for specific shell commands seems like a reasonable compromise.

This can be dealt with and discussed further in #57. Making the window caption/title configurable should be its own ticket, although it arguably depends on #57 too.

**Comment by eXeC64** [comment]
Resolved by the addition of custom config files + key bindings
